File: medplum-java/medplum-java-model/src/main/java/com/fhir/TestReport_Operation.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableTestReport_Operation.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface TestReport_Operation {
  Optional<String> id();

  Optional<uri> detail();

  Optional<List<Extension>> extension();

  Optional<List<Extension>> modifierExtension();

  Optional<markdown> message();

  Optional<Testreport_operationResult> result();

  static ImmutableTestReport_Operation.Builder builder() {
    return ImmutableTestReport_Operation.builder();
  }
}

File: medplum-typescript/src/model/fhir/TestReport_Operation.ts
import { Extension } from "./Extension";
import { markdown } from "./markdown";
import { Testreport_operationResult } from "./Testreport_operationResult";
import { uri } from "./uri";

export interface TestReport_Operation {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    result: Testreport_operationResult | null | undefined;
    message: markdown | null | undefined;
    detail: uri | null | undefined;
}
